base/iterators.jl

```diff
@@ -1595,4 +1595,72 @@ end
 # be the same as the keys, so this is a valid optimization (see #51631)
 pairs(s::AbstractString) = IterableStatePairs(s)
 
+"""
+    nth(itr, n::Integer)
+
+Get the `n`th element of an iterable collection. Return `nothing` if not existing.
+
+See also: [`first`](@ref), [`last`](@ref)
+
+# Examples
+```jldoctest
+julia> nth(2:2:10, 4)
+8
+
+julia> nth(reshape(1:30, (5,6)), 6)
+6
+```
+"""
+nth(itr, n::Integer) = _nth(IteratorSize(itr), itr, n)
+nth(itr::AbstractArray, n::Integer) = n > length(itr) ? nothing : itr[n]
```

Comment:
This assumes one-based indexing. Perhaps do `itr[begin + n - 1]`.

Reply:
you are absolutely correct.
would something like `getindex(itr, nth(eachindex(IndexLinear(), itr), n))` be too overkill?
and adding a specialization with `nth(itr::AbstractRange, n::Integer) = getindex(itr, n)`

Reply:
I went with the probably overkill approach, if it's too much i'll revert back to your suggestion.

Reply:
`AbstractRange`s are not always one-based either, so that approach runs into the same issue

Reply:
From what I could gather that is included in the `getindex` already, since it ends up calling
```julia
unsafe_getindex(v::AbstractRange{T}, i::Integer) where T = convert(T, first(v) + (i - oneunit(i))*step_hp(v))
```
unless I'm missing the point completely?

Reply:
This line `nth(itr::AbstractRange, n) = getindex(itr, n)` will for sure fail on the axes of an OffsetArray.

Reply:
I was overthinking it, I'll just stick with `[begin + n - 1]`. Sorry.